index.js

```diff
@@ -47,6 +50,60 @@ function fastifyWebsocket (fastify, opts, next) {
   const wss = new WebSocket.Server(wssOptions)
   fastify.decorate('websocketServer', wss)
 
+  async function injectWS (path = '/', upgradeContext = {}) {
+    const server2Client = new PassThrough()
+    const client2Server = new PassThrough()
+
+    const serverStream = new Duplexify(server2Client, client2Server)
+    const clientStream = new Duplexify(client2Server, server2Client)
+
+    const ws = new WebSocket(null, { isServer: false })
+    const head = Buffer.from([])
+
+    let resolve, reject
+    const promise = new Promise((_resolve, _reject) => { resolve = _resolve; reject = _reject })
+
+    ws.on('open', () => {
+      clientStream.removeListener('data', onData)
+      resolve(ws)
+    })
+
+    const onData = (chunk) => {
+      if (chunk.toString().includes('HTTP/1.1 101 Switching Protocols')) {
+        ws._isServer = false
+        ws.setSocket(clientStream, head, { maxPayload: 0 })
+      } else {
+        clientStream.removeListener('data', onData)
+
+        reject(new Error('Websocket injection failed'))
```

Comment:
@airhorns Can you give me some feedback about this? 
I want to give more information to the user on why it fails, but I have no idea how to include it.

This is an example of the chunk I am getting.
```
HTTP/1.1 401 Unauthorized
content-length: 0
Date: Wed, 08 Nov 2023 17:45:54 GMT
Connection: keep-alive
```

Reply:
Sorry, I don't quite understand the questIon! That chunk makes sense to me, your hook in the test replies with a 401 which means the websocket would never be initiated, right?

Reply:
Indeed, but the user sees only 'Websocket injection failed'. 

I want to be more accurate and describe the reason, maybe including a part of the chunk.

Reply:
What happens when you construct a websocket in a normal node or browser context to an endpoint that 401s? I think the behaviour should be as close to that as possible.